mockall_derive/src/mock_function.rs

```diff
@@ -988,13 +988,17 @@ impl<'a> ToTokens for Common<'a> {
                 fn drop(&mut self) {
                     if !::std::thread::panicking() && !self.times.is_satisfied()
                     {
-                        let desc = std::format!(
-                            "{}", self.matcher.lock().unwrap());
-                        panic!("{}: Expectation({}) called {} time(s) which is fewer than expected {}",
-                               #funcname,
-                               desc,
-                               self.times.count(),
-                               self.times.minimum());
+                        if (self.times.count() < self.times.minimum()) {
```

Comment:
This is very confusing because you're checking times.count() here, and also implicitly up above in is_satisfied.  It makes the code hard to read.  You should only do that check on one level.

Reply:
Could you elaborate on this point?

Now is_satisfied checks both expected boundaries of the actual number of calls and returns a boolean.
From a boolean I'm not able to determine what error message to print in the panic, this is why the second check.

Instead of a boolean I could return an Enum with values: SATISFIED_LOWER_BOUND, SATISFIED_UPPER_BOUND, SATISFIED, NOT_SATISFIED. 
Then I could differentiate the cases. But I don't know if this is what you had in mind...

Reply:
You could do something like that, returning an enum.  Or you could inline the is_satisfied logic into the proc macro.  Either could work.  I just don't want that logic to be repeated in two places.